lib/client/client.go

```diff
@@ -198,7 +212,13 @@ func (proxy *ProxyClient) reissueUserCerts(ctx context.Context, params ReissuePa
 	key := params.ExistingCreds
 	if key == nil {
 		var err error
-		key, err = proxy.localAgent().GetKey(params.RouteToCluster, WithAllCerts...)
+		// Note that we are explicitly NOT persisting any existing service
+		// (e.g. DB, Kubernetes, etc) certificates across the re-issue of the
+		// main certificate. This is because the re-issued certificate may
+		// contain new roles, and preserving the old certificates will
+		// only stop those new roles being applied to the service
+		// certificate.
+		key, err = proxy.localAgent().GetKey(params.RouteToCluster)
```

Comment:
Just to avoid any misunderstanding: This call has nothing to do with *persisting* the key data in the store itself. It's meant just for the convenience of the caller in case they happen to make use of k8s clusters/DBs/apps not mentioned in the reissue request itself.

Reply:
Removed during refactor